shlr/tcc/tccgen.c

```diff
@@ -3182,8 +3192,15 @@ static int decl0(int l, int is_for_loop_init) {
 					char buf[500];
 					alias = get_tok_str(v, NULL);
 					type_to_str(buf, sizeof(buf), &sym->type, NULL);
-					tcc_appendf("%s=typedef\n",alias);
-					tcc_appendf("typedef.%s=%s\n",alias, buf);
+					tcc_appendf ("%s=typedef\n", alias);
+					tcc_appendf ("typedef.%s=%s\n", alias, buf);
+					if (tcc_typedefs) {
+						while (*tcc_typedefs != NULL) { // XXX it should not enter here
+							tcc_appendf (*tcc_typedefs, alias);
```

Comment:
segfaults here

Reply:
@radare I don't really understand why this happens. Isn't the condition above checking if the pointer holds valid data? should I use calloc to allocate tcc_typedefs?

Reply:
the != NULL is unnecessary just `while (*tcc_typedefs)`